src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigAlignmentSignatureClassifier.java

```diff
@@ -0,0 +1,171 @@
+
+        // split between strand switch or not (NOTE BOTH SAME CHROMOSOME MAPPING)
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> strandSwitchOrNot =
+                RDDUtils.split(sameChrOrNot._1, AssemblyContigAlignmentSignatureClassifier::indicatesIntraChrStrandSwitchBkpts, false);
+        contigsByRawTypes.put(RawTypes.IntraChrStrandSwitch, strandSwitchOrNot._1);
+
+        // split between ref block switch or not (NOTE BOTH SAME CHROMOSOME MAPPING AND NO STRAND SWITCH)
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> tandemDupBkptOrSimpleInsDel =
+                RDDUtils.split(strandSwitchOrNot._2, AssemblyContigAlignmentSignatureClassifier::indicatesIntraChrTandemDupBkpts, false);
+        contigsByRawTypes.put(RawTypes.InsDel, tandemDupBkptOrSimpleInsDel._2);
+        contigsByRawTypes.put(RawTypes.TandemDupOrMEIBkpt, tandemDupBkptOrSimpleInsDel._1);
+    }
+
+    //==================================================================================================================
+
+    // TODO: 10/27/17 add tests for these predicates
+
+    @VisibleForTesting
+    static boolean hasIncompletePictureDueToRefSpanContainment(final AlignedContig contigWithOnlyOneConfigAnd2Aln) {
+        Utils.validateArg(hasOnly2Alignments(contigWithOnlyOneConfigAnd2Aln),
+                "assumption that input contig has only 2 alignments is violated. \n" +
+                        contigWithOnlyOneConfigAnd2Aln.toString());
+        final AlignmentInterval one = contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(0),
+                                two = contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(1);
+        return one.referenceSpan.contains(two.referenceSpan)
+                ||
+                two.referenceSpan.contains(one.referenceSpan);
+    }
+
+    /**
+     * @param contigWithOnlyOneConfigAnd2Aln assumed to have only two alignments
+     */
+    @VisibleForTesting
+    static boolean isSameChromosomeMapping(final AlignedContig contigWithOnlyOneConfigAnd2Aln) {
+        Utils.validateArg(!hasIncompletePictureDueToRefSpanContainment(contigWithOnlyOneConfigAnd2Aln),
+                "assumption that input contig has alignments whose ref span is not completely covered by the other's is violated \n" +
+                        contigWithOnlyOneConfigAnd2Aln.toString());
+        return contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(0).referenceSpan.getContig()
+                .equals(contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(1).referenceSpan.getContig());
+    }
+
+    /**
+     * @param contigWithOnlyOneConfigAnd2AlnToSameChr assumed to have only two alignments mapped to the same chromosome
+     */
+    @VisibleForTesting
+    static boolean indicatesIntraChrStrandSwitchBkpts(final AlignedContig contigWithOnlyOneConfigAnd2AlnToSameChr) {
+        Utils.validateArg(isSameChromosomeMapping(contigWithOnlyOneConfigAnd2AlnToSameChr),
```

Comment:
Given the method names, maybe the right thing to do here is to just return false if the predicate doesn't hold? Ie return false if the mappings are on different chromosomes.

Reply:
done